### PR TITLE
E2E testing

### PR DESCRIPTION
**Pull request by JasonStoltz**
This Adds E2E Tests in Kibana using the Functional Test Runner. The Functional Test Runner is the Kibana Operations team's preferred method of implementing E2E tests; it uses Selenium, WebDriver, and Mocha. It includes a number of tools for running and interacting with the Elastic Stack.

- [x] Get tests wired up
- [x] Get App Search wired up to tests
- ~~Get App Search to come up with FTR~~
- [x] Figure out how to log in as enterprise_search user so that Authentication works
- [x] Create App Search Service to load fixtures for tests
- [x] Write tests
- [x] Write README for running tests

The commands for running these tests are included in the README. Please follow through the README and ensure that these tests actually work.

There are only a few tests but they do hit some key areas:
1. Configuration - It's hard to test configuration via a config file in Unit Tests, so this adds coverage there.
2. Happy Path - Ensures that a user can successfully navigate to the plugin and see engine and meta engines.

There are some more in depth notes included here: https://docs.google.com/document/d/1X3cGsnGalCInPMNggO6q6qk89nZLUO2poxsl9Jxac8s/edit#, for reference.

This PR essentially gets the basics wired up. We could discuss what we'd be interested in doing next. For instance:

A) Add more E2E tests
B) Work on actually getting this to spin Enterprise Search up before running, which I think is doable.
C) Do nothing, this is enough E2E test coverage for now, plus we may be interesting in abandoning this for Cypress in the future.

Closes #991

### CODE REVIEW

File: x-pack/test/functional/apps/enterprise_search/index.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function({ loadTestFile }: FtrProviderContext) {
+  describe('Enterprise Search', function() {
+    loadTestFile(require.resolve('./setup_guide'));
+  });
+}
```
Comment (cee-chen) [inline x-pack/test/functional/apps/enterprise_search/index.ts]: The difference in file directory locations confuses me a decent amount for these two different suites. Is there any particular reason for that, or can we put both of these under `x-pack/test/functional/apps/` and name one `enterprise_search` and `enterprise_search_nonconfigured` or `enterprise_search_setup` perhaps?

Comment (JasonStoltz) [reply x-pack/test/functional/apps/enterprise_search/index.ts]: Good point, making an updated now to put them in 

test/functional_enterprise_search/apps/enterprise_search/with_host_configured
test/functional_enterprise_search/apps/enterprise_search/without_host_configured

The original idea was to have the test that didn't require the host configuration to run with the rest of the functional suite.

File: x-pack/test/functional_enterprise_search/apps/enterprise_search/with_host_configured/app_search/engines.ts [hunk 2]
```diff
@@ -33,12 +35,28 @@ export default function enterpriseSearchSetupEnginesTests({
...
+        await PageObjects.security.login('enterprise_search', 'changeme', {
+          expectSpaceSelector: false,
+        });
+
         await PageObjects.enterpriseSearch.navigateToPage();
         await retry.try(async function() {
           const currentUrl = await browser.getCurrentUrl();
           expect(currentUrl).to.contain('/app_search');
         });
+        browser.getActions().pause(5000);
```
Comment (richkuz) [inline x-pack/test/functional_enterprise_search/apps/enterprise_search/with_host_configured/app_search/engines.ts]: Is there a more concrete condition we can wait for here?

Comment (JasonStoltz) [reply x-pack/test/functional_enterprise_search/apps/enterprise_search/with_host_configured/app_search/engines.ts]: Sorry, I checked this in in a premature state to show my progress.

I'm not relying on that, I had put that in temporarily in an attempt to inspect the page in chrome manually.

Comment (richkuz) [reply x-pack/test/functional_enterprise_search/apps/enterprise_search/with_host_configured/app_search/engines.ts]: 👍 And, in hindsight, my minor comment was likely premature for a Draft PR. I'll try to keep Draft PR feedback more focused on the big picture.

Comment (JasonStoltz) [reply x-pack/test/functional_enterprise_search/apps/enterprise_search/with_host_configured/app_search/engines.ts]: No no, I'm actually glad you dropped that comment. I think it's acceptable to look at a Draft PR and notice concerning code. Like it's probably better to catch things like that early before it gets out of hand. Like if I started added pauses like that all over the code it would be a pretty terrible decision.

File: x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_overview/engine_overview.tsx [hunk 3]
```diff
@@ -104,7 +104,7 @@ export const EngineOverview: ReactFC<> = () => {
               </h2>
             </EuiTitle>
           </EuiPageContentHeader>
-          <EuiPageContentBody>
+          <EuiPageContentBody data-test-subj="appSearchEngines">
```
Comment (JasonStoltz) [inline x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_overview/engine_overview.tsx:111]: Kibana recommends using `data-test-subj` for finding elements in tests and even has a `testSubjects` [service](https://www.elastic.co/guide/en/kibana/current/development-functional-tests.html#_kibana_services) helper that makes it easier to find elements by this attribute.

File: x-pack/test/functional_enterprise_search/apps/enterprise_search/with_host_configured/app_search/engines.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    describe('when an enterpriseSearch host is configured', () => {
+      it('navigating to the enterprise_search plugin will redirect a user to the App Search Engines Overview page', async () => {
+        await PageObjects.security.forceLogout();
+        const { user, password } = appSearch.getEnterpriseSearchUser();
+        await PageObjects.security.login(user, password, {
+          expectSpaceSelector: false,
+        });
+
+        await PageObjects.appSearch.navigateToPage();
+        await retry.try(async function() {
```
Comment (JasonStoltz) [inline x-pack/test/functional_enterprise_search/apps/enterprise_search/with_host_configured/app_search/engines.ts]: The `retry` service lets us retry a particular command until it succeeds.

File: x-pack/test/functional_enterprise_search/services/app_search_client.ts [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import http from 'http';
+
+/**
+ * A simple request client for making API calls to the App Search API
+ */
+const makeRequest = <T>(method: string, path: string, body?: object): Promise<T> => {
```
Comment (JasonStoltz) [inline x-pack/test/functional_enterprise_search/services/app_search_client.ts:12]: I created a simple client to use here to avoid bringing the App Search api client in as a new dependency to Kibana.

File: x-pack/test/functional_enterprise_search/with_host_configured.config.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
+/*
```
Comment (JasonStoltz) [inline x-pack/test/functional_enterprise_search/with_host_configured.config.ts:1]: Every time you run a test with the functional test runner, you need to specify which configuration file to use.

Kibana configuration can be set in these files. Configuration like `enterpriseSearch.host`.

Because of this, we need 2 separate files for testing when `enterpriseSearch.host` is set or not, which is why we also need two test commands, as noted in the README.

Both test files extend a common "base" config file for shared concerns.

File: x-pack/test/functional_enterprise_search/README.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+from that instance set in an Environment variable.
+
+Ex.
+
+```
+# Run specs from the x-pack directory
+cd x-pack
+
+# Run tests that require enterpriseSearch.host variable
+APP_SEARCH_API_KEY=private-2shg47w6zpj62zqe8qb5fap9 node scripts/functional_tests --config test/functional_enterprise_search/with_host_configured.config.ts
```
Comment (cee-chen) [inline x-pack/test/functional_enterprise_search/README.md]: Are we concerned at all about having this API key in a publicly-exposed document? I'm not sure if it's real or fake, but if fake, can we at least make it something like `private-12345`?

If it's meant to be a private key from the currently-spun-up local App Search instance, can we make that clearer via something like:

`APP_SEARCH_API_KEY=[use private key from local App Search instance here]`

~Although I'm also actually not totally clear what we need the API key for?...~ 🤦‍♀️  disregard sorry, makes sense that it's for creating engines (until we can do so via the plugin itself)

Comment (cee-chen) [reply x-pack/test/functional_enterprise_search/README.md]: OK after testing myself locally, looks like it's supposed to be a real private key from local AS. I'd suggest this:

```suggestion
APP_SEARCH_API_KEY=[use private key from local App Search instance here] node scripts/functional_tests --config test/functional_enterprise_search/with_host_configured.config.ts
```

File: x-pack/test/functional_enterprise_search/README.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+## Enterprise Search Requirement
+
+These tests will not currently start an instance of App Search automatically. As such, they are not run as part of CI and are most useful for local regression testing.
+
+Requirements for Enterprise Search:
+
+- Running on port 3002 against a separate Elasticsearch cluster.
+- Elasticsearch must have a platinum or greater level license (or trial).
+- Must have Standard Auth configured with an `enterprise_search` user with password `changeme`.
```
Comment (cee-chen) [inline x-pack/test/functional_enterprise_search/README.md]: Elasticsearch Native auth also works!
```suggestion
- Must have Standard or Native Auth configured with an `enterprise_search` user with password `changeme`.
```

File: x-pack/test/functional_enterprise_search/README.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+```
+
+## Enterprise Search Requirement
+
+These tests will not currently start an instance of App Search automatically. As such, they are not run as part of CI and are most useful for local regression testing.
+
+Requirements for Enterprise Search:
+
+- Running on port 3002 against a separate Elasticsearch cluster.
+- Elasticsearch must have a platinum or greater level license (or trial).
```
Comment (cee-chen) [inline x-pack/test/functional_enterprise_search/README.md:39]: Note to self - if missing a >platinum license, you'll get a rather mysterious error of `SyntaxError: Unexpected end of JSON input`

File: x-pack/test/functional/page_objects/enterprise_search.ts [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+ */
+
+import { FtrProviderContext } from '../ftr_provider_context';
+
+export function EnterpriseSearchPageProvider({ getService, getPageObjects }: FtrProviderContext) {
+  const PageObjects = getPageObjects(['common']);
+
+  return {
+    async navigateToPage() {
+      return await PageObjects.common.navigateToApp('enterprise_search');
```
Comment (cee-chen) [inline x-pack/test/functional/page_objects/enterprise_search.ts]: Just curious, how will this be affected (if at all) when we're registering 2-3 separate plugins (App Search, Workplace Search, and Enterprise Search)?

(This is mostly just a far-future question/me wondering out loud)

File: x-pack/test/functional_enterprise_search/page_objects/index.ts [hunk 11]
```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { pageObjects as basePageObjects } from '../../functional/page_objects';
```
Comment (cee-chen) [inline x-pack/test/functional_enterprise_search/page_objects/index.ts:7]: (This is not necessarily a change request since we appear to be reusing existing architecture, just me thinking out loud)

Maybe I'm missing something... but why put our page objects in `x-pack/test/functional/page_objects` when we could just put them in `x-pack/test/functional_enterprise_search/page_objects` instead?

File: x-pack/test/functional_enterprise_search/page_objects/app_search.ts [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+    async navigateToPage() {
+      return await PageObjects.common.navigateToApp('app_search');
+    },
+
+    async getEngineLinks() {
+      const engines = await testSubjects.find('appSearchEngines');
+      return await testSubjects.findAllDescendant('engineNameLink', engines);
+    },
+
+    async getMeatEngineLinks() {
```
Comment (cee-chen) [inline x-pack/test/functional_enterprise_search/page_objects/app_search.ts]: Ahahaha I'm so sorry, this cracked me up but
```suggestion
    async getMetaEngineLinks() {
```
Now I'm thinking real hard about what a meat engine would look like 😂

File: x-pack/test/functional_enterprise_search/apps/enterprise_search/with_host_configured/app_search/engines.ts [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+          const currentUrl = await browser.getCurrentUrl();
+          expect(currentUrl).to.contain('/app_search');
+        });
+      });
+
+      it('lists engines', async () => {
+        const engineLinks = await PageObjects.appSearch.getEngineLinks();
+        const engineLinksText = await Promise.all(engineLinks.map(l => l.getVisibleText()));
```
Comment (cee-chen) [inline x-pack/test/functional_enterprise_search/apps/enterprise_search/with_host_configured/app_search/engines.ts]: Super interesting syntax - definitely very different from Cypress, that's for sure!

File: x-pack/test/functional_enterprise_search/apps/enterprise_search/with_host_configured/app_search/engines.ts [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+
+        expect(engineLinksText.includes(engine1.name)).to.equal(true);
+        expect(engineLinksText.includes(engine2.name)).to.equal(true);
+      });
+
+      it('lists meta engines', async () => {
+        const metaEngineLinks = await PageObjects.appSearch.getMeatEngineLinks();
```
Comment (cee-chen) [inline x-pack/test/functional_enterprise_search/apps/enterprise_search/with_host_configured/app_search/engines.ts]:
```suggestion
        const metaEngineLinks = await PageObjects.appSearch.getMetaEngineLinks();
```
😂

File: x-pack/test/functional_enterprise_search/page_objects/app_search.ts [hunk 15]
```diff
@@ -0,0 +1,29 @@
...
+
+  return {
+    async navigateToPage() {
+      return await PageObjects.common.navigateToApp('app_search');
+    },
+
+    async getEngineLinks() {
+      const engines = await testSubjects.find('appSearchEngines');
+      return await testSubjects.findAllDescendant('engineNameLink', engines);
+    },
```
Comment (cee-chen) [inline x-pack/test/functional_enterprise_search/page_objects/app_search.ts:22]: So, I've kinda had this conversation before but for Cypress, but I get mildly concerned with (what I think is?) over-DRYing stuff out instead of just using the native APIs inline.

1. It requires someone new to the testing framework API to jump between 2 different files to understand what the helper is actually doing
2. If we continue to use a `getSomeSelector()` helper for every single new test we add, this file is going to quickly balloon in size and become increasingly difficult to parse/read

This isn't a change request but more a general philosophy that but I'd prefer to keep selector helpers in the file they're used in, and only start DRYing them out to a shared file once more than 2-3+ tests duplicate them.

File: x-pack/test/functional/page_objects/enterprise_search.ts (hunk 10 above)
Comment (cee-chen) [reply x-pack/test/functional/page_objects/enterprise_search.ts]: Ahh disregard my question here, I see that we use [`.navigateToApp('app_search')`](https://github.com/constancecchen/kibana/pull/6/files#diff-a61f4f16fea7f280127bf23d9430acfcR15)

File: x-pack/test/functional_enterprise_search/services/app_search_client.ts [hunk 16]
```diff
@@ -0,0 +1,114 @@
...
+  return new Promise(function(resolve, reject) {
+    // Throw error if no key is configured
+    // Handle API errors
```
Comment (cee-chen) [inline x-pack/test/functional_enterprise_search/services/app_search_client.ts]: Is `// Handle API errors` a stray comment that got accidentally moved? I'm a little confused as to how it applies directly to the line directly below it 🤔

File: x-pack/test/functional_enterprise_search/services/app_search_client.ts [hunk 17]
```diff
@@ -0,0 +1,114 @@
...
+        isReady = true;
+        resolve();
+      }
+    }
+  });
+};
+
+export const indexData = async (engineName: string, docs: object[]) => {
+  return await makeRequest('POST', `/api/as/v1/engines/${engineName}/documents`, docs);
+};
```
Comment (cee-chen) [inline x-pack/test/functional_enterprise_search/services/app_search_client.ts:121]: Just wanted to say this is a super impressive client - huge kudos! 👏 

Do you envision that we'll be able to potentially phase it out at some point and be able to complete all API actions within the Kibana UI naturally? (the same way / what we do right now for Cypress tests in AS?)

File: x-pack/test/functional_enterprise_search/services/app_search_service.ts [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+  IEngine,
+} from './app_search_client';
+
+export interface IUser {
+  user: string;
+  password: string;
+}
+export { IEngine };
+
+export class AppSearchService {
```
Comment (cee-chen) [inline x-pack/test/functional_enterprise_search/services/app_search_service.ts:26]: I'm just curious - I feel like I've almost grasped this but would also love a 'human-readable' explanation so to speak. What differentiates the concept of the App Search **Client** from the App Search **Service**? I get that the client manages the nitty-gritty http calls - is the Service just a thin and easier-to-read wrapper on top of that?

File: x-pack/test/functional_enterprise_search/services/app_search_client.ts (hunk 16 above)
Comment (JasonStoltz) [reply x-pack/test/functional_enterprise_search/services/app_search_client.ts]: I think that was supposed to say "// TODO Handle API errors"

File: x-pack/test/functional_enterprise_search/services/app_search_client.ts (hunk 17 above)
Comment (JasonStoltz) [reply x-pack/test/functional_enterprise_search/services/app_search_client.ts:121]: I think if we ever had the ability to create engines from within the UI, then yes, then I would imagine we could phase this out.

I also wondered about using the `esArchiver` to load fixtures into ES for our various tests, which would be pretty neat.

File: x-pack/test/functional_enterprise_search/services/app_search_service.ts (hunk 18 above)
Comment (JasonStoltz) [reply x-pack/test/functional_enterprise_search/services/app_search_service.ts:26]: Services are a first class concept in Kibana: https://www.elastic.co/guide/en/kibana/current/development-functional-tests.html#_custom_services. Having an App Search service to encapsulate all of the setup / interaction with App Search seemed ideal.

Splitting out the low level API interactions to a client just cleans up the Service.

I'm primarily using the Service like a Factory in Rails https://github.com/elastic/ent-search/blob/master/spec/factories/engines.rb, creating different types of test fixtures... createComplexEngine, createSimpleEngine, createPokemonEngine, etc.

File: x-pack/test/functional_enterprise_search/page_objects/app_search.ts (hunk 12 above)
Comment (JasonStoltz) [reply x-pack/test/functional_enterprise_search/page_objects/app_search.ts]: That was totally intentional 🍖 .

File: x-pack/test/functional_enterprise_search/page_objects/app_search.ts (hunk 15 above)
Comment (JasonStoltz) [reply x-pack/test/functional_enterprise_search/page_objects/app_search.ts:22]: That's a fair point, thanks.

File: x-pack/test/functional_enterprise_search/apps/enterprise_search/with_host_configured/app_search/engines.ts (hunk 13 above)
Comment (JasonStoltz) [reply x-pack/test/functional_enterprise_search/apps/enterprise_search/with_host_configured/app_search/engines.ts]: There may be a cleaner way in there to do that, I'm not sure. 

The fact that `getVisibleText()` returns a Promise required the Promise.all...  I do wonder if it supports calling `.getVisibleText()` on the collection?

Comment (JasonStoltz) [reply x-pack/test/functional_enterprise_search/apps/enterprise_search/with_host_configured/app_search/engines.ts]: I'm not familiar enough with Cypress so I'm not sure what the equivalent would look like.

File: x-pack/test/functional_enterprise_search/README.md (hunk 9 above)
Comment (JasonStoltz) [reply x-pack/test/functional_enterprise_search/README.md:39]: Ugh, yeah. If we decide we want to keep these tests around I think I may just drop a docker-compose file in that will bring up the stack configured correctly.

Comment (JasonStoltz) [reply x-pack/test/functional_enterprise_search/README.md:39]: So you don't even have to think about this stuff.

File: x-pack/test/functional_enterprise_search/page_objects/index.ts (hunk 11 above)
Comment (JasonStoltz) [reply x-pack/test/functional_enterprise_search/page_objects/index.ts:7]: I think I mean to delete that one from `pack/test/functional/page_objects`, I thought I did anyway. Let me clean that up, because I agree with you.

File: x-pack/test/functional_enterprise_search/services/app_search_client.ts (hunk 16 above)
Comment (JasonStoltz) [reply x-pack/test/functional_enterprise_search/services/app_search_client.ts]: I'm going to address this.

File: x-pack/test/functional_enterprise_search/services/app_search_service.ts (hunk 18 above)
Comment (cee-chen) [reply x-pack/test/functional_enterprise_search/services/app_search_service.ts:26]: Ahhh, thanks, that helped clarify that for me!

**Review by cee-chen — APPROVED** [review]
LGTM, although let me know if you'd rather get in fixes for the one or two remaining comments first!

File: x-pack/test/functional_enterprise_search/services/app_search_client.ts (hunk 16 above)
Comment (JasonStoltz) [reply x-pack/test/functional_enterprise_search/services/app_search_client.ts]: Added error handling so that when this thing bombs out on an App Search API call, it gives you enough context to fix it.

```
     Error: the string "Error calling App Search API: {\"error\":\"Invalid authentication token.\"}" was thrown, throw an Error :)


       └- ✖ fail: "Enterprise Search Engines Overview "before all" hook in "Engines Overview""
```

File: x-pack/test/functional_enterprise_search/page_objects/index.ts (hunk 11 above)
Comment (JasonStoltz) [reply x-pack/test/functional_enterprise_search/page_objects/index.ts:7]: Done: https://github.com/constancecchen/kibana/pull/6

**Comment by JasonStoltz** [comment]
Hey @constancecchen, this PR should be ready to go. Sorry for the delay in responding to your feedback.

I suggest squash merging this PR, there are many thrashing commits and poorly worded commit messages.

File: x-pack/test/functional_enterprise_search/page_objects/app_search.ts (hunk 12 above)
Comment (cee-chen) [inline x-pack/test/functional_enterprise_search/page_objects/app_search.ts]: Did some changes get reverted? This is back to being `getMeatEngine` (vs `Meta`)

File: x-pack/test/functional_enterprise_search/README.md (hunk 7 above)
Comment (cee-chen) [reply x-pack/test/functional_enterprise_search/README.md]: It looks like this change was also reverted at some point

File: x-pack/test/functional_enterprise_search/README.md (hunk 8 above)
Comment (cee-chen) [reply x-pack/test/functional_enterprise_search/README.md]: It looks like this change was also reverted at some point

File: x-pack/test/functional_enterprise_search/services/app_search_client.ts (hunk 16 above)
Comment (cee-chen) [reply x-pack/test/functional_enterprise_search/services/app_search_client.ts]: Awesome, looks terrific!

**Comment by cee-chen** [comment]
It looks like a few changes got reverted somehow - I unresolved those convos. Everything else LGTM though once that's addressed!

**Comment by cee-chen** [comment]
Doh, not sure if you've seen this as well, but I'm getting `ERROR SessionNotCreatedError: session not created: This version of ChromeDriver only supports Chrome version 81` when trying to run the tests locally.

My local Chrome is on version 83 so I think maybe that's why?...

**Comment by JasonStoltz** [comment]
@constancecchen Looks like `"chromedriver": "^83.0.0"`, is correct in `package.json, try kbn bootstrap.

**Comment by JasonStoltz** [comment]
I rebased to your 7.x branch

**Comment by JasonStoltz** [comment]
@constancecchen That's so weird, not sure how they got reverted. I'm assuming I screwed up during a rebase though. I think this is good to go now, finally.

File: x-pack/test/functional_enterprise_search/README.md [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+
+```
+# Run specs from the x-pack directory
+cd x-pack
+
+# Run tests that require enterpriseSearch.host variable
+APP_SEARCH_API_KEY=[use private key from local App Search instance here] node scripts/functional_tests --config test/functional_enterprise_search/with_host_configured.config.ts
+
+# Run tests that do not require enterpriseSearch.host variable
+node scripts/functional_tests --config test/functional_enterprise_search/without_host_configured.config.ts
```
Comment (cee-chen) [inline x-pack/test/functional_enterprise_search/README.md]: I'm running this and getting a `Please provide a valid APP_SEARCH_API_KEY. See README for more details.` error in my terminal - any advice?

File: x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_overview/engine_table.tsx [hunk 20]
```diff
@@ -56,7 +56,11 @@ export const EngineTable: ReactFC<IEngineTableProps> = ({
       name: i18n.translate('xpack.appSearch.enginesOverview.table.column.name', {
         defaultMessage: 'Name',
       }),
-      render: name => <EuiLink {...engineLinkProps(name)}>{name}</EuiLink>,
+      render: name => (
```
Comment (cee-chen) [inline x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_overview/engine_table.tsx]:
```suggestion
      render: (name) => (
```

I think some Prettier settings recently changed, I had to do an eslint pass and it caught some things that needed changing

**Review by cee-chen — COMMENTED** [review]
Very minor linting asks (since the proj will fail the Kibana's lint test otherwise) - can you run the following commands from Kibana root:

1. `yarn prettier --write x-pack/test/functional_enterprise_search/` (fixes Prettier lint issues)
2. `yarn lint:es x-pack/test/functional_enterprise_search/` (just to check that there are no issues, looks like there aren't though)

Thanks!

File: x-pack/test/functional_enterprise_search/README.md (hunk 19 above)
Comment (cee-chen) [reply x-pack/test/functional_enterprise_search/README.md]: I ran this with the `APP_SEARCH_API_KEY=[xxx]` prefix as in the above example and the test passed after that.

No strong feelings either way, but I'd suggest either removing the req for `APP_SEARCH_API_KEY` for the without_host config, or updating this command to include it.

Comment (cee-chen) [reply x-pack/test/functional_enterprise_search/README.md]: BTW - can I ask that we add `bash` right after the ``` code blocks in this markdown file? That way all the code snippets get syntax highlighting :)

Comment (JasonStoltz) [reply x-pack/test/functional_enterprise_search/README.md]: Yes, good call

**Comment by JasonStoltz** [comment]
I thought I had my editor configured to use Kibana's lint settings but I guess it's not working as I thought. Thanks for catching the lint fix.

**Comment by cee-chen** [comment]
Check if your local Prettier instance/IDE is using Prettier 2.0.5 - Prettier changed some default settings in 2.x and I was on 1.9, so for me that's what caused linting mismatches

**Comment by JasonStoltz** [comment]
I think I just wrote that code before I had fixed my editor settings. It looks good now.

**Review by cee-chen — APPROVED** [review]
🎉

**Comment by richkuz** [comment]
Good collaboration on this PR, really thorough review and implementation. 👍 👊
